### PR TITLE
fix to enable load.balancing option

### PR DESCRIPTION
**Pull request by helenaK**
It was not possible to enable load.balancing option for the following functions even if load.balancing was set to TRUE:
parallelStartMulticore
parallelStartSocket
parallelStartMPI
Modifications (line 206 - 223) are independent from batchtools

### CODE REVIEW

**Comment by berndbischl** [comment]
thx.

1) you seem to have batchtools code in there? which we dont supprt yet?

2) also pls always at least briefly describe what goes on in the PR in the PR thread!

**Comment by berndbischl** [comment]
hi

1) travis fails, but this cannot really be due to your code. i will check this.

2) do you have quick local example code snippet so i can ensure that load balancing works?
i would like to have a unit test for this, but i guess this is hard. 
but i would at least like to "see" that it works now

File: R/parallelStart.R [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #'   you want processes on multiple machines use \code{socket.hosts}.
 #'   Default is the option \code{parallelMap.default.cpus} or, if not set,
 #'   \code{\link[parallel]{detectCores}} for multicore mode,
-#'   \code{max(1, \link[Rmpi]{mpi.universe.size} - 1)} for mpi mode
```
Comment (jakob-r) [inline R/parallelStart.R]: Why this change? I think it is correct.

Comment (helenaK) [inline R/parallelStart.R]: I changed it back. Travis still fails.

**Comment by jakob-r** [comment]
@berndbischl I checked this and it failed because we never actually tests load balancing. `parallel::clusterApplyLB` does not have the arguments `SIMPLIFY, USE.NAMES` unlike `clusterMap`. So I left them out. I think the behaviour of `clusterApplyLB` is identical to `SIMPLIFY = FALSE` and `USE.NAMES = TRUE` but I don't know of `USE.NAMES` has an effect in our case.

**Comment by jakob-r** [comment]
The Windows build seems to fail because of `backports` missing. Can somebody have a look at this?

**Comment by mllg** [comment]
AppVeyor was not installing all dependencies. This should be fixed now upstream.

**Comment by jakob-r** [comment]
I can't see any changes on master. Can you please restart Appveyor and/or merge?

**Comment by berndbischl** [comment]
> Can you please restart Appveyor and/or merge?

i have asked multiple things here in the thread which were simply ignored. i would like to have them answered before i merge

**Comment by jakob-r** [comment]
A1) Maybe by mistake. Not anymore since a while
A2) load.balancing was not properly passed to `parallelStart` and as a results always deactivated

B1) Travis does not fail anymore
B2) `partest7` was added wich exactly tests that. One long job and muliple short jobs. Overall runtime should not be longer then the one of the long job.

**Comment by jakob-r** [comment]
*ping*
And why does AppVeyor still fail? (@mllg?)

**Comment by mllg** [comment]
Push something to trigger a rebuild. I don't have a button for this.

**Comment by berndbischl** [comment]
@mllg do you see any problems here?
i browsed through it and i guess i can merge

@jakob-r thx for the reply. sorry, for taking longer, holiday

**Comment by mllg** [comment]
Thanks, merged now.
